drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -54,56 +54,71 @@ static inline void msg_init(struct device *dev, struct i2c_msg *msg,
 	}
 }
 
-static inline void msg_done(struct device *dev, unsigned int current_msg_flags)
-{
-	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
-	I2C_TypeDef *i2c = cfg->i2c;
-
-	/* Wait for transfer to complete */
-	while (!LL_I2C_IsActiveFlag_TC(i2c) && !LL_I2C_IsActiveFlag_TCR(i2c)) {
-		;
-	}
-	/* Issue stop condition if necessary */
-	if (current_msg_flags & I2C_MSG_STOP) {
-		LL_I2C_GenerateStopCondition(i2c);
-		while (!LL_I2C_IsActiveFlag_STOP(i2c)) {
-			;
-		}
-		LL_I2C_ClearFlag_STOP(i2c);
-		LL_I2C_DisableReloadMode(i2c);
-	}
-}
-
 #ifdef CONFIG_I2C_STM32_INTERRUPT
 void stm32_i2c_event_isr(void *arg)
 {
 	const struct i2c_stm32_config *cfg = DEV_CFG((struct device *)arg);
 	struct i2c_stm32_data *data = DEV_DATA((struct device *)arg);
 	I2C_TypeDef *i2c = cfg->i2c;
 
-	if (data->current.is_write) {
-		if (data->current.len && LL_I2C_IsActiveFlag_TXIS(i2c)) {
+	/* Interrupts for sending/receiving next byte */
+	if (data->current.len) {
+
+		/* Send next byte */
+		if (data->current.is_write && LL_I2C_IsActiveFlag_TXIS(i2c)) {
 			LL_I2C_TransmitData8(i2c, *data->current.buf);
+
+		/* Receive next byte */
+		} else if (!(data->current.is_write)
+				   && LL_I2C_IsActiveFlag_RXNE(i2c)) {
```

Comment:
Move && on upper line and align LL_I2C on (data->

Reply:
@ erwango thanks for all the comments!
I haven't found anything in the linux kernel coding style descriptions on how exactly the alignment in multi line if statements should look.
You wrote 
>and align LL_I2C on (data->

After moving it to the next line, it looks as if it should be aligned to !(data-> like this
```c
} else if (!(data->current.is_write) && 
           LL_I2C_IsActiveFlag_RXNE(i2c)) {
```
not to (data->

What is your opinion on this? 
Can you point me to any doc of the linux coding style, that is more specific on this topic than [coding-style.rst](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/coding-style.rst) from the linux kernel itself?

Reply:
You're right! Align on "!". I read erroneously that "!" applied to whole condition, which is not the case.
One benefit of this "logical based alignment" is that it help avoiding this kind of quick reading mistake.
I haven't found explicit reference to this, but seems commonly used in zephyr (and I see it implicitly described in GNU emacs proposed configuration in "9)  You’ve made a mess of it" in https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/coding-style.rst)